### PR TITLE
Clone Datasette's method for handling python environments

### PR DESCRIPTION
**Pull request by sfc-gh-zblackwood**

### CODE REVIEW

File: download_python.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+release_date="20221106"
+filename="cpython-3.8.15+20221106-aarch64-apple-darwin-install_only.tar.gz"
```
Comment (sfc-gh-tsimons) [inline download_python.sh:3]: great choice -- I made a branch also and chose this one!

File: main.js [hunk 2]
```diff
@@ -127,8 +128,10 @@ Menu.setApplicationMenu(menu)
 const TOP = 30;
 const TOTAL_WIDTH = 2000;
 const TOTAL_HEIGHT = 1000;
-const LEFT_WIDTH = 1200;
+const LEFT_WIDTH = 1000;
```
Comment (sfc-gh-tsimons) [inline main.js:131]: I was looking to see if there's a way to get the width automatically for the user's screen size. I'm not sure exactly but I thought something like 
`const electron = require('electron'), const remote = electron.remote, remote.getCurrentWindow().webContents.getOwnerBrowserWindow().getBounds()`
could work.

File: python_environment.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+const mkdir = util.promisify(fs.mkdir)
+const cp = require("child_process")
+const execFile = util.promisify(cp.execFile)
+const EventEmitter = require("events");
+const { dialog } = require("electron")
+
+
+
+const minPackageVersions = {
+  "streamlit": "1.15.2",
```
Comment (sfc-gh-tsimons) [inline python_environment.js:17]: 🎈

**Review by sfc-gh-tsimons — APPROVED** [review]
Looks good -- clearly it works as Amber is able to get it running. My env is having more difficulty.

File: download_python.sh (hunk 1 above)
Comment (sfc-gh-zblackwood) [reply download_python.sh:3]: Yeah, I copied this script from your branch :)

File: main.js (hunk 2 above)
Comment (sfc-gh-zblackwood) [reply main.js:131]: I'll have to give that a try!

**Pull request merged**